test/TestProjects/MgmtPropertyChooser/MgmtPropertyChooser.json

```diff
@@ -0,0 +1,556 @@
+                }
+            },
+            "description": "Describes the properties of a Virtual Machine."
+        },
+        "VirtualMachineExtension": {
+            "properties": {
+                "properties": {
+                    "x-ms-client-flatten": true,
+                    "$ref": "#/definitions/VirtualMachineExtensionProperties"
+                }
+            },
+            "allOf": [
+                {
+                    "$ref": "#/definitions/Resource"
+                }
+            ],
+            "description": "Describes a Virtual Machine Extension."
+        },
+        "VirtualMachineExtensionProperties": {
+            "properties": {
+                "forceUpdateTag": {
+                    "type": "string",
+                    "description": "How the extension handler should be forced to update even if the extension configuration has not changed."
+                },
+                "publisher": {
+                    "type": "string",
+                    "description": "The name of the extension handler publisher."
+                },
+                "type": {
+                    "type": "string",
+                    "description": "Specifies the type of the extension; an example is \"CustomScriptExtension\"."
+                },
+                "typeHandlerVersion": {
+                    "type": "string",
+                    "description": "Specifies the version of the script handler."
+                },
+                "autoUpgradeMinorVersion": {
+                    "type": "boolean",
+                    "description": "Indicates whether the extension should use a newer minor version if one is available at deployment time. Once deployed, however, the extension will not upgrade minor versions unless redeployed, even with this property set to true."
+                },
+                "enableAutomaticUpgrade": {
+                    "type": "boolean",
+                    "description": "Indicates whether the extension should be automatically upgraded by the platform if there is a newer version of the extension available."
+                },
+                "settings": {
+                    "type": "object",
+                    "description": "Json formatted public settings for the extension."
+                },
+                "protectedSettings": {
+                    "type": "object",
+                    "description": "The extension can contain either protectedSettings or protectedSettingsFromKeyVault or no protected settings at all."
+                },
+                "provisioningState": {
+                    "readOnly": true,
+                    "type": "string",
+                    "description": "The provisioning state, which only appears in the response."
+                }
+            },
+            "description": "Describes the properties of a Virtual Machine Extension."
+        },
+        "VirtualMachineIdentity": {
```

Comment:
Can we add another model that shouldn't be replaced?  Like what if I have this exact thing with myPrincipalId for the property name?  Should that be picked up?  What about principalId being an int vs a string?